.github/workflows/wheels-build.yaml

```diff
@@ -112,6 +118,7 @@ jobs:
   build:
     name:  ${{ matrix.CUDA_VER }}, ${{ matrix.PY_VER }}, ${{ matrix.ARCH }}, ${{ matrix.LINUX_VER }} 
     needs: [compute-matrix]
+    continue-on-error: ${{ matrix.CUDA_VER == '12.2.2' }}
```

Comment:
> ### Why not a branch from a fork?
> So I can test workflows in other repos like rmm and cudf using the workflows from this branch.

FYI -- if we're committed to trying the "continue on error" approach, it won't matter if this PR comes from a fork. We would merge it to the upstream `rapidsai/shared-workflows`, and just let the jobs fail. The only time it matters if it comes from a fork is for using branch name updates in each repo's actions that we revert after completing the migration.

Reply:
However, it gets a little weird for this case of updating CUDA 12.2 if we were to merge this PR with `continue-on-error`, because we would like to test the artifacts from both 12.0 (the status quo) and 12.2 (the new version) but the artifacts will have the same name `-cu12`. I'm not sure how we would distinguish this while making the updates non-breaking. For example, dropping CUDA 12.0 builds at the same time would be breaking if the new 12.2 build jobs don't pass CI, and it wouldn't even make sense to do `continue-on-error`. We'd have to add builds for 12.2 that _don't_ upload artifacts, make sure they pass, and then disable 12.0 at the same time we make the 12.2 builds upload their artifacts. Unless there's a way to change the names of the 12.2 artifacts and test those explicitly...? This needs some consideration.

Reply:
Just checking myself on this -- I probably need to refresh myself on how we determine which wheel artifacts to publish. We can build for 12.2 and not publish, and we can test the 12.0 builds on 12.2 drivers. I just don't know how that works in practice because I am not 100% sure whether this step would run into conflicting names from the 12.0 wheels and 12.2 wheels: https://github.com/rapidsai/shared-workflows/blob/4384050a538188eb6f4d19cb3865cdecb86b3c46/.github/workflows/wheels-build.yaml#L181

Reply:
>  if we're committed to trying the "continue on error" approach, it won't matter if this PR comes from a fork

This makes sense to me in general.

I wanted to do this from a branch here *this time* because I'm also testing my use of the `continue-on-error` mechanism itself. I was nervous about misunderstanding that mechanism and merging something that breaks CI in a merge-blocking way for a bunch of projects all at once.

> This needs some consideration.

Definitely, let me think on that! 🤔 

I forgot that the wheels will have package names (and therefore filenames) like `{project}-cu{major}`.

Reply:
> I am not 100% sure whether this step would run into conflicting names from the 12.0 wheels and 12.2 wheels

Your intuition was right... it would run into conflicts.

That statement you linked to trims out the first `.` and everything after it.

```shell
RAPIDS_CUDA_VERSION="12.2.2"
echo "${RAPIDS_CUDA_VERSION%%.*}"
# 12
```

The CUDA *major* version is baked into artifact names and not used to disambiguate objects in S3 (see https://github.com/rapidsai/ops/issues/2982#issuecomment-1861157155 for reference).

e.g. see https://downloads.rapids.ai/nightly/cudf/2024-01-04/4c01e95/

<img width="843" alt="Screenshot 2024-01-04 at 11 49 51 AM" src="https://github.com/rapidsai/shared-workflows/assets/7608904/b125419f-4b28-46be-a226-6b48738278cf">

So as this is all currently set up, having `build-*` steps with different CUDA major versions would result in artifacts conflicting.

I'm thinking about this 🤔

Reply:
Individual projects are also uploading and downloading artifacts from code inside repo-local scripts.

For example, from `rmm`:

```yaml
  wheel-build:
    needs: checks
    secrets: inherit
    uses: rapidsai/shared-workflows/.github/workflows/wheels-build.yaml@branch-24.02
    with:
      build_type: pull-request
      script: ci/build_wheel.sh
```

([pr.yaml](https://github.com/rapidsai/rmm/blob/e2dc7279d9a6bed2860697bcb5155a0ee6e57efe/.github/workflows/pr.yaml#L65-L71))

```shell
package_name="rmm"

# ...

RAPIDS_PY_CUDA_SUFFIX="$(rapids-wheel-ctk-name-gen ${RAPIDS_CUDA_VERSION})"

# ...

RAPIDS_PY_WHEEL_NAME="${package_name}_${RAPIDS_PY_CUDA_SUFFIX}" rapids-upload-wheels-to-s3 final_dist
```

([build_wheel.sh](https://github.com/rapidsai/rmm/blob/e2dc7279d9a6bed2860697bcb5155a0ee6e57efe/ci/build_wheel.sh#L39))

Reply:
We decided to address this by abandoning the `continue-on-error` approach and instead doing the "open PRs on all the RAPIDS repos" approach. That seemed simpler than messing around with artifact-naming.

See the PR description for more details.